scene/3d/baked_lightmap.cpp

```diff
@@ -1254,14 +1254,25 @@ Ref<Image> BakedLightmap::_get_irradiance_from_sky(Ref<Sky> p_sky, Vector2i p_si
 
 	sky_image->convert(Image::FORMAT_RGBF);
 	sky_image->resize(p_size.x, p_size.y, Image::INTERPOLATE_CUBIC);
+
+	if (p_energy != 1.0) {
```

Comment:
With floating point numbers, should this be a tolerance test?
`if (abs(p_energy - 1.0) > TOLERANCE)`

Reply:
The check is only a shortcut optimization when the energy value is exactly 1, since `x * 1 = x`. Any other value would give a different result.